draft-gazdag-x509-hash-sigs.md

```diff
@@ -666,57 +617,44 @@ This ASN.1 Module builds upon the conventions established in [RFC5911].
-would have at least 128 bits of quantum collision-resistance as well as the
-pre-image resistance mentioned in the previous paragraph.
-
-Given the quantum collision and pre-image resistance of SHA-256 estimated
-above, the current parameter sets used by `id-alg-hss-lms-hashsig`,
-`id-alg-xmss-hashsig` and `id-alg-xmssmt-hashsig` provide 128 bits or more of
-quantum security. This is believed to be secure enough to protect X.509
-certificates for well beyond any reasonable certificate lifetime.
-
-## Implementation Security Considerations
-
-Implementations MUST protect the private keys. Compromise of the private keys
-may result in the ability to forge signatures. Along with the private key, the
-implementation MUST keep track of which leaf nodes in the tree have been used.
-Loss of integrity of this tracking data can cause a one-time key to be used
-more than once. As a result, when a private key and the tracking data are
-stored on non- volatile media or stored in a virtual machine environment, care
-must be taken to preserve confidentiality and integrity.
-
-The generation of private keys relies on random numbers. The use of inadequate
-pseudo-random number generators (PRNGs) to generate these values can result in
-little or no security. An attacker may find it much easier to reproduce the
-PRNG environment that produced the keys, searching the resulting small set of
-possibilities, rather than brute force searching the whole key space. The
-generation of quality random numbers is difficult. [RFC4086] offers important
-guidance in this area.
-
-The generation of hash-based signatures also depends on random numbers. While
-the consequences of an inadequate pseudo-random number generator (PRNG) to
-generate these values is much less severe than the generation of private keys,
-the guidance in [RFC4086] remains important.
+The security requirements of [NIST-SP-800-208] and [NIST-FIPS-205] MUST be
+taken into account.
+
+For the stateful HBS (HSS, XMSS, XMSS^MT) it is crucial to stress the
+importance of a correct state management. If an attacker were able to obtain
+signatures for two different messages created using the same OTS key, then it
+would become computationally feasible for that attacker to create forgeries
+[BH16]. As noted in [MCGREW], extreme care needs to be taken in order to avoid
+the risk that an OTS key will be reused accidentally. This is a new requirement
+that most developers will not be familiar with and requires careful handling.
+
+Various strategies for a correct state management can be applied:
+
+- Implement a track record of all signatures generated by a key pair associated
+  to a stateful HBS instance. This track record may be stored outside the
+  device which is used to generate the signature. Check the track record to
+  prevent OTS key reuse before a new signature is released. Drop the new
+  signature and hit your PANIC button if you spot OTS key reuse.
```

Comment:
I suggest to also put in the state reservation mechanism too. It is explained in section 5 of [MCGREW].

Reply:
Added to the list of strategies. See https://github.com/x509-hbs/draft-gazdag-x509-hash-sigs/pull/12/commits/610f419cd21d94a5eb8c0bd5dce90e5512d91289